backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/monitoring/VirtMonitoringStrategy.java

```diff
@@ -45,19 +50,29 @@ public class VirtMonitoringStrategy implements MonitoringStrategy {
     @Inject
     private Instance<IVdsEventListener> eventListener;
 
+    @Inject
+    private ImageTransferDao imageTransferDao;
+
     @Inject
     public VirtMonitoringStrategy(ClusterDao clusterDao,
             VdsDao vdsDao,
             VmDao vmDao,
-            VmDynamicDao vmDynamicDao) {
+            VmDynamicDao vmDynamicDao,
+            ImageTransferDao imageTransferDao) {
         this.clusterDao = clusterDao;
         this.vdsDao = vdsDao;
         this.vmDao = vmDao;
         this.vmDynamicDao = vmDynamicDao;
+        this.imageTransferDao = imageTransferDao;
     }
 
     @Override
     public boolean canMoveToMaintenance(VDS vds) {
+        List<ImageTransfer> transfers = imageTransferDao.getByVdsId(vds.getId());
+        if (!transfers.stream().allMatch(ImageTransfer::isPausedOrFinished)) {
+            log.debug("Waiting fo all image transfers to be finished");
```

Comment:
1. IMHO the message is confusing/not clear, suggest to rephrase:
`Can't move to maintenance, there are active image transfers`.
2. If you added a log message here, maybe worth also adding them (instead of the existing comments) in 2 other cases? @ahadas what do you think?
`Can't move to maintenance, there are VMs running on the host`.
`Can't move to maintenance, there are managed non-external VMs running on the host `.

Reply:
Changed it to "Can't move to maintenance, there are VMs running on the host"

Reply:
Looks good,can resolve the comment :)